unsupported/Eigen/CXX11/src/Tensor/TensorReductionGpu.h

```diff
@@ -295,17 +281,6 @@ __global__ void FullReductionKernelHalfFloat(Reducer reducer, const Self input,
     atomicReduce(scratch, accum, reducer);
   }
 
-#if defined(EIGEN_HIPCC)
-  __syncthreads();
```

Comment:
@deven-amd same here. could you explain the rationale why HIP-specific logic is not needed here anymore?